### PR TITLE
Include platformType and lastUpdated in CDx model & Use JPA join fetch for eager fetch

### PR DESCRIPTION
**Pull request by zhx828**
This is related to https://github.com/oncokb/oncokb/issues/3565
This fixes https://github.com/oncokb/oncokb/issues/3576

### CODE REVIEW

File: src/main/webapp/app/entities/companion-diagnostic-device/companion-diagnostic-device.tsx [hunk 1]
```diff
@@ -50,53 +52,63 @@ export const CompanionDiagnosticDevice = (props: ICompanionDiagnosticDeviceProps
 
   useEffect(() => {
     if (search) {
-      debouncedSearch(search, props.searchEntities);
+      // debouncedSearch(search, props.searchEntities);
```
Comment (zhx828) [inline src/main/webapp/app/entities/companion-diagnostic-device/companion-diagnostic-device.tsx:55]: This needs to be updated once merge.

File: src/main/webapp/app/shared/util/pagination-crud-store.ts [hunk 2]
```diff
@@ -4,14 +4,19 @@ import BaseCrudStore from 'app/shared/util/base-crud-store';
...
-  (query, page, size, sort, searchEntities) =>
-    searchEntities({
-      query,
-      page,
-      size,
-      sort,
-    }),
-  500,
+  (query: string, page, size, sort, searchEntities) => {
+    if (query && query.length > 2) {
```
Comment (calvinlu3) [inline src/main/webapp/app/shared/util/pagination-crud-store.ts:8]: We could show a message to user that search has to be at least 3 characters. I can look into this in my PR to replace OncoKbTable in transcript with public.

File: src/main/java/org/mskcc/oncokb/curation/web/rest/DrugController.java [hunk 3]
```diff
@@ -1,37 +0,0 @@
...
-        log.debug("REST request to search Drugs");
-        return drugService.searchDrug(query);
-    }
-
-    @GetMapping("/drugs/search-by-code/{code}")
-    public Optional<Drug> findDrugByCode(@PathVariable(value = "code") String code) {
-        log.debug("REST request to search Drugs");
-        return drugService.findByCode(code);
-    }
-}
```
Comment (calvinlu3) [inline src/main/java/org/mskcc/oncokb/curation/web/rest/DrugController.java:37]: Just a note that this is used in oncokb-core. When we merge rc branch into master in transcript, we need to update the usage in the core to use `DrugResource.searchDrugs()`

File: src/main/webapp/app/entities/drug/drug.store.ts [hunk 4]
```diff
@@ -5,17 +5,17 @@ import PaginationCrudStore from 'app/shared/util/pagination-crud-store';
...
   constructor(protected rootStore: IRootStore) {
     super(rootStore, apiUrl);
   }
-  *getSearch({ query, page, size, sort }) {
-    const result = yield axios.get<IDrug[]>(`${apiSearchUrl}?query=${query}${sort ? `&page=${page}&size=${size}&sort=${sort}` : ''}`);
+  *getSearch({ query }) {
+    const result = yield axios.get<IDrug[]>(`${apiSearchUrl}?query=${query}`);
     this.entities = result.data;
-    this.totalItems = result.headers['x-total-count'];
+    this.totalItems = result.data.length();
```
Comment (calvinlu3) [inline src/main/webapp/app/entities/drug/drug.store.ts:18]:
```suggestion
    this.totalItems = result.data.length;
```

**Review by calvinlu3 — CHANGES_REQUESTED** [review]

File: src/main/java/org/mskcc/oncokb/curation/web/rest/DrugController.java (hunk 3 above)
Comment (zhx828) [reply src/main/java/org/mskcc/oncokb/curation/web/rest/DrugController.java:37]: Added an issue and updated the task list in https://github.com/oncokb/oncokb/issues/3465

**Review by calvinlu3 — APPROVED** [review]

**Pull request merged**